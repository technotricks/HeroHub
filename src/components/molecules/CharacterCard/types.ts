import {ItemClickCallback} from '@/types/base';
import {Character} from '@/types/character';

export type CardProps = {
  item: Character;
  onPress: ItemClickCallback;
};
