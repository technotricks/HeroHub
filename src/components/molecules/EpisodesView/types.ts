import {CharacterDetail} from '@/types/character';

export type EpisodesViewProps = {
  item?: CharacterDetail;
};
