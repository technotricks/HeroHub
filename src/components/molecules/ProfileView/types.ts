import {CharacterDetail} from '@/types/character';

export type ProfileViewProps = {
  item?: CharacterDetail;
};
