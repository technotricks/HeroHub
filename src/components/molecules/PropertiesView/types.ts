import {CharacterDetail} from '@/types/character';

export type PropertiesViewProps = {
  item?: CharacterDetail;
};
