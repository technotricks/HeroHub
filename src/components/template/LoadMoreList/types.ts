import {ItemClickCallback, LoadMoreCallback} from '@/types/base';
import {Character} from '@/types/character';

export namespace ILoadMore {
  export type Props = {
    items: Character[];
    isLoading: Boolean;
    onPress: ItemClickCallback;
    handleLoadMore: LoadMoreCallback;
  };
}
