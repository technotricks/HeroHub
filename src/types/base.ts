import {Character} from '@/types/character';

export type ItemClickCallback = (item: Character) => void;
export type LoadMoreCallback = () => void;
