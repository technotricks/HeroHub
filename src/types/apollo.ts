export type QueryResult<T> = {
  loading: boolean;
  data: T | undefined;
  error: Error | undefined;
};

export type Variables = {
  [key: string]: any;
};

export type QueryHookOptions = {
  fetchPolicy?:
    | 'cache-first'
    | 'cache-and-network'
    | 'network-only'
    | 'cache-only';
  variables?: Variables;
};
