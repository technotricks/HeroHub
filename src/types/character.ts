export interface Character {
  id: number;
  name: string;
  status: string;
  image: string;
  gender: string;
  type: string;
}

type NameObject = {
  name: string;
};

type Episode = {
  name: string;
  episode: string;
};

export interface CharacterDetail extends Character {
  origin: NameObject;
  location: NameObject;
  episode: Episode[];
}

type Info = {
  id: number;
  name: number;
  status: number;
};

// Create a TypeScript type for the query result
export type GetCharactersData = {
  characters: {
    results: Character[];
    info: Info;
  };
};

// Create a TypeScript type for the query result
export type GetCharacterData = {
  characters: {
    results: CharacterDetail[];
  };
};

export type FetchCharacters = (page: number) => void;
export type FetchCharacter = (id: number) => void;
