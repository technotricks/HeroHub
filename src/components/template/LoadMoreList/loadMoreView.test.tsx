import 'react-native';
import React from 'react';

import {it, expect, describe} from '@jest/globals';

// Note: test renderer must be required after react-native.
import {act, create} from 'react-test-renderer';
import LoadMoreView from '.';
import {Character} from '@/types/character';

describe('LoadMoreView component should render correctly', () => {
  const characters: Character[] = [
    {
      id: 1,
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
      name: 'Rick Sanchez',
      status: 'Alive',
      gender: 'Male',
      type: '',
    },
    {
      id: 2,
      image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
      name: 'Morty Smith',
      status: 'Alive',
      gender: 'Male',
      type: '',
    },
    {
      id: 3,
      image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg',
      name: 'Summer Smith',
      status: 'Alive',
      gender: 'Female',
      type: '',
    },
  ];
  const mockCallBack = jest.fn();

  const view = create(
    <LoadMoreView items={characters} isLoading={true} onPress={mockCallBack} />,
  );

  it('snapshot', () => {
    expect(view).toMatchSnapshot();
  });
});
