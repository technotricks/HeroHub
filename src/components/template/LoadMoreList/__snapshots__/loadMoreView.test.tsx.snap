// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMoreView component should render correctly snapshot 1`] = `
<RCTScrollView
  ListFooterComponent={
    <ActivityIndicator
      color="red"
      size="large"
    />
  }
  data={
    [
      {
        "gender": "Male",
        "id": 1,
        "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        "name": "Rick Sanchez",
        "status": "Alive",
        "type": "",
      },
      {
        "gender": "Male",
        "id": 2,
        "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
        "name": "Morty Smith",
        "status": "Alive",
        "type": "",
      },
      {
        "gender": "Female",
        "id": 3,
        "image": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
        "name": "Summer Smith",
        "status": "Alive",
        "type": "",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={0.1}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="buttonID"
          >
            <Image
              source={
                {
                  "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 100,
                  "width": "100%",
                }
              }
              testID="imageID"
            />
            <Text
              aligin="center"
              color="#FF87A2"
              fontSize="18px"
              isBold={true}
              style={
                {
                  "color": "#FF87A2",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
              testID="nameID"
            >
              Rick Sanchez
            </Text>
            <Text
              aligin="left"
              color="#2CB9B0"
              fontSize="16px"
              isBold={true}
              style={
                {
                  "color": "#2CB9B0",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "left",
                }
              }
              testID="statusID"
            >
              Alive
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="buttonID"
          >
            <Image
              source={
                {
                  "uri": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 100,
                  "width": "100%",
                }
              }
              testID="imageID"
            />
            <Text
              aligin="center"
              color="#FF87A2"
              fontSize="18px"
              isBold={true}
              style={
                {
                  "color": "#FF87A2",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
              testID="nameID"
            >
              Morty Smith
            </Text>
            <Text
              aligin="left"
              color="#2CB9B0"
              fontSize="16px"
              isBold={true}
              style={
                {
                  "color": "#2CB9B0",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "left",
                }
              }
              testID="statusID"
            >
              Alive
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="buttonID"
          >
            <Image
              source={
                {
                  "uri": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 100,
                  "width": "100%",
                }
              }
              testID="imageID"
            />
            <Text
              aligin="center"
              color="#FF87A2"
              fontSize="18px"
              isBold={true}
              style={
                {
                  "color": "#FF87A2",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
              testID="nameID"
            >
              Summer Smith
            </Text>
            <Text
              aligin="left"
              color="#2CB9B0"
              fontSize="16px"
              isBold={true}
              style={
                {
                  "color": "#2CB9B0",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "left",
                }
              }
              testID="statusID"
            >
              Alive
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <ActivityIndicator
        color="red"
        size="large"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`PropertiesView component should render correctly snapshot 1`] = `
<RCTScrollView
  ListFooterComponent={
    <ActivityIndicator
      color="red"
      size="large"
    />
  }
  data={
    [
      {
        "gender": "Male",
        "id": 1,
        "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
        "name": "Rick Sanchez",
        "status": "Alive",
        "type": "",
      },
      {
        "gender": "Male",
        "id": 2,
        "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
        "name": "Morty Smith",
        "status": "Alive",
        "type": "",
      },
      {
        "gender": "Female",
        "id": 3,
        "image": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
        "name": "Summer Smith",
        "status": "Alive",
        "type": "",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={0.1}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="buttonID"
          >
            <Image
              source={
                {
                  "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 100,
                  "width": "100%",
                }
              }
              testID="imageID"
            />
            <Text
              aligin="center"
              color="#FF87A2"
              fontSize="18px"
              isBold={true}
              style={
                {
                  "color": "#FF87A2",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
              testID="nameID"
            >
              Rick Sanchez
            </Text>
            <Text
              aligin="left"
              color="#2CB9B0"
              fontSize="16px"
              isBold={true}
              style={
                {
                  "color": "#2CB9B0",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "left",
                }
              }
              testID="statusID"
            >
              Alive
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="buttonID"
          >
            <Image
              source={
                {
                  "uri": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 100,
                  "width": "100%",
                }
              }
              testID="imageID"
            />
            <Text
              aligin="center"
              color="#FF87A2"
              fontSize="18px"
              isBold={true}
              style={
                {
                  "color": "#FF87A2",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
              testID="nameID"
            >
              Morty Smith
            </Text>
            <Text
              aligin="left"
              color="#2CB9B0"
              fontSize="16px"
              isBold={true}
              style={
                {
                  "color": "#2CB9B0",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "left",
                }
              }
              testID="statusID"
            >
              Alive
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#ffffff",
              "borderBottomLeftRadius": 10,
              "borderBottomRightRadius": 10,
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="buttonID"
          >
            <Image
              source={
                {
                  "uri": "https://rickandmortyapi.com/api/character/avatar/3.jpeg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 10,
                  "borderBottomRightRadius": 10,
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "height": 100,
                  "width": "100%",
                }
              }
              testID="imageID"
            />
            <Text
              aligin="center"
              color="#FF87A2"
              fontSize="18px"
              isBold={true}
              style={
                {
                  "color": "#FF87A2",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
              testID="nameID"
            >
              Summer Smith
            </Text>
            <Text
              aligin="left"
              color="#2CB9B0"
              fontSize="16px"
              isBold={true}
              style={
                {
                  "color": "#2CB9B0",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "textAlign": "left",
                }
              }
              testID="statusID"
            >
              Alive
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <ActivityIndicator
        color="red"
        size="large"
      />
    </View>
  </View>
</RCTScrollView>
`;
